apps/cli/src/server/sync.ts

```diff
@@ -0,0 +1,95 @@
+import * as ADCSDK from '@api7/adc-sdk';
+import type { RequestHandler } from 'express';
+import { omit, toString } from 'lodash';
+import { lastValueFrom, toArray } from 'rxjs';
+
+import { loadBackend } from '../command/utils';
+import { DifferV3 } from '../differ/differv3';
+import { check } from '../linter';
+import { SyncInput, type SyncInputType } from './schema';
+
+export const syncHandler: RequestHandler<
+  unknown,
+  unknown,
+  SyncInputType
+> = async (req, res) => {
+  const parsedInput = SyncInput.safeParse(req.body);
+  if (!parsedInput.success)
+    return res.status(400).json({
+      message: parsedInput.error.message,
+      errors: parsedInput.error.issues,
+    });
+  const { task } = parsedInput.data;
+
+  // load local configuration and validate it
+  const local = task.config as ADCSDK.Configuration;
+  if (task.opts.lint) {
+    const result = check(local);
+    if (!result.success)
+      return res.status(400).json({
+        message: result.error.message,
+        errors: result.error.issues,
+      });
+  }
+
+  try {
+    // load remote configuration and diff with local
+    const backend = loadBackend(task.opts.backend, { ...task.opts });
+    const remote = await lastValueFrom(backend.dump());
+    const diff = DifferV3.diff(local, remote, await backend.defaultValue());
+
+    // sync the diff
+    const results = await lastValueFrom(
+      backend
+        .sync(diff, {
+          exitOnFailure: false,
+          concurrent: 4,
```

Comment:
I'm not sure if a race condition issue will happen here, causing the unexpected behavior.

Reply:
The backend is responsible for enforcing the concurrent policy, which is only valid for API7 and APISIX. They have correctly and fully taken this into account, i.e., staged concurrent requests by resource priority.